File: src/com/esprit/entities/Rewards.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.esprit.entities;
/**
 *
 * @author oumayma
 */
public class Rewards {
     private int id;
      private int rank;
      private String trophe;
       private float prixR;
        private  String img;
        private Tournoi t;

    public Rewards(int id, String img) {
        this.id = id;
        this.img = img;
    }

    public Rewards(int id) {
        this.id = id;
    }

    public Rewards(int id, Tournoi t) {
        this.id = id;
        this.t = t;
    }

    public Rewards(Tournoi t) {
        this.t = t;
    }

    public Rewards() {
    }

    public Rewards(String img) {
       
        this.img = img;
    }

    public Rewards(int rank, String trophe, float prixR, String img, Tournoi t) {
        this.rank = rank;
        this.trophe = trophe;
        this.prixR = prixR;
        this.img = img;
        this.t = t;
    }

    public Rewards( int rank, String trophe, float prixR, String img) {
   
        this.rank = rank;
        this.trophe = trophe;
        this.prixR = prixR;
        this.img = img;
    }

    public Rewards(int id, int rank, String trophe, float prixR, String img, Tournoi t) {
        this.id = id;
        this.rank = rank;
        this.trophe = trophe;
        this.prixR = prixR;
        this.img = img;
        this.t = t;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getRank() {
        return rank;
    }

    public void setRank(int rank) {
        this.rank = rank;
    }

    public String getTrophe() {
        return trophe;
    }

    public void setTrophe(String trophe) {
        this.trophe = trophe;
    }

    public float getPrixR() {
        return prixR;
    }

    public void setPrixR(float prixR) {
        this.prixR = prixR;
    }

    public String getImg() {
        return img;
    }

    public void setImg(String img) {
        this.img = img;
    }

    public Tournoi getT() {
        return t;
    }

    public void setT(Tournoi t) {
        this.t = t;
    }
       
      
    
}

File: src/Entity/Rewards.php
<?php

namespace App\Entity;

use App\Repository\RewardsRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Component\Serializer\Annotation\Groups;
/**
 * @ORM\Entity(repositoryClass=RewardsRepository::class)
 */
class Rewards
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     *  @Groups("s")
     */
    private $idR;

    /**
     * @ORM\Column(type="integer")
     *  @Assert\NotBlank
     * @Groups("s")
     */
    private $rank;

    /**
     * @ORM\Column(type="string", length=255)
     *  @Assert\NotBlank(message="veillez remplir le champ")
     * @Groups("s")
     */
    private $trophe;

    /**
     * @ORM\Column(type="float")
     *  * @Assert\NotBlank(message="veillez entrer un prix")
     * @Groups("s")
     */
    private $prixR;



    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Upload your image")
     * @Groups("s")
     */
    private $img;

    /**
     * @ORM\ManyToOne(targetEntity=Tournoi::class, inversedBy="rewards")
     * @ORM\JoinColumn(nullable=false)
     * @Groups("s")
     */
    private $tournoi;

    public function getIdR(): ?int
    {
        return $this->idR;
    }

    public function getRank(): ?int
    {
        return $this->rank;
    }

    public function setRank(int $rank): self
    {
        $this->rank = $rank;

        return $this;
    }

    public function getTrophe(): ?string
    {
        return $this->trophe;
    }

    public function setTrophe(string $trophe): self
    {
        $this->trophe = $trophe;

        return $this;
    }

    public function getPrixR(): ?float
    {
        return $this->prixR;
    }

    public function setPrixR(float $prixR): self
    {
        $this->prixR = $prixR;

        return $this;
    }



    public function getImg(): ?string
    {
        return $this->img;
    }

    public function setImg(string $img): self
    {
        $this->img = $img;

        return $this;
    }

    public function getTournoi(): ?Tournoi
    {
        return $this->tournoi;
    }

    public function setTournoi(?Tournoi $tournoi): self
    {
        $this->tournoi = $tournoi;

        return $this;
    }
}